### PR TITLE
feat: add `HR` component

### PR DESCRIPTION
**Pull request by dhavalveera**
Implemented the HR component from Flowbite package into our Flowbite React project. This enhancement empowers users to conveniently HR for the divider. The integration seamlessly aligns with Flowbite's design principles and ensures consistency across the project.

## Summary by CodeRabbit

- **New Features**
  - Introduced a new `HR` component for customizable horizontal rules with subcomponents for icons, text, and square styles.
  - Added documentation for creating and customizing horizontal lines using the new `HR` component with Tailwind CSS.
  - Updated the documentation sidebar to include the new `HR` component for easier navigation and access.

### CODE REVIEW

File: apps/web/content/docs/components/hr.mdx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+---
+title: React Horizontal Line (HR) - Flowbite
+description: Create a horizontal line using the HR tag to separate content such as paragraphs, blockquotes, and other elements using the utility classes from Tailwind CSS
+---
+
+The HR component can be used to separate content using a horizontal line by adding space between elements based on multiple styles, variants, and layouts.
+
+## Default HR
+
+Use this example to separate text content with a <hr> horizontal line.
```
Comment (SutuSebastian) [inline apps/web/content/docs/components/hr.mdx:10]: try this for the `typecheck` CI error:

```md
Use this example to separate text content with a `<hr>` horizontal line.
```

File: packages/ui/src/components/HR/HRIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+}
+
+export const HRIcon = forwardRef<HTMLHRElement, HRIconProps>(
+  ({ theme: customTheme = {}, icon: Icon, className, ...props }, ref) => {
+    const theme = mergeDeep(getTheme().hr.icon, customTheme);
+
+    const SVGIcon = Icon ? Icon : FaQuoteRight;
+
+    return (
+      <>
```
Comment (SutuSebastian) [inline packages/ui/src/components/HR/HRIcon.tsx:30]: redundant fragment

File: packages/ui/src/components/HR/HRText.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  text: string;
+  theme?: DeepPartial<FlowbiteHRTextTheme>;
+}
+
+export const HRText = forwardRef<HTMLHRElement, HRTextProps>(
+  ({ theme: customTheme = {}, text, className, ...props }, ref) => {
+    const theme = mergeDeep(getTheme().hr.text, customTheme);
+
+    return (
+      <>
```
Comment (SutuSebastian) [inline packages/ui/src/components/HR/HRText.tsx:24]: redundant fragment

File: apps/web/data/docs-sidebar.ts [hunk 4]
```diff
@@ -67,6 +67,7 @@ export const DOCS_SIDEBAR: DocsSidebarSection[] = [
       { title: "Dropdown", href: "/docs/components/dropdown" },
       { title: "Footer", href: "/docs/components/footer" },
       { title: "Forms", href: "/docs/components/forms" },
+      { title: "HR", href: "/docs/components/hr", isNew: true },
```
Comment (rluders) [inline apps/web/data/docs-sidebar.ts:70]: This should be actually under `typography` section

**Review by rluders — CHANGES_REQUESTED** [review]

File: apps/web/data/docs-sidebar.ts (hunk 4 above)
Comment (dhavalveera) [reply apps/web/data/docs-sidebar.ts:70]: Gotcha!.. I wasn't aware of it, so added it here... Will Change it,

File: apps/web/content/docs/components/hr.mdx [hunk 5]
```diff
@@ -0,0 +1,46 @@
+---
```
Comment (rluders) [inline apps/web/content/docs/components/hr.mdx:1]: This file should be also moved to https://github.com/themesberg/flowbite-react/tree/main/apps/web/content/docs/typography

**Review by rluders — CHANGES_REQUESTED** [review]

File: apps/web/content/docs/components/hr.mdx (hunk 5 above)
Comment (dhavalveera) [reply apps/web/content/docs/components/hr.mdx]: Fixed!.

**Comment by rluders** [comment]
@SutuSebastian storybook is broken here, do you know why?

**Comment by SutuSebastian** [comment]
> @SutuSebastian storybook is broken here, do you know why?

checking

**Comment by dhavalveera** [comment]
@SutuSebastian - can you please review?

**Pull request merged**